include/alpaka/vec/Vec.hpp

```diff
@@ -511,8 +511,7 @@ namespace alpaka
         }
 
     private:
-        // 16 Byte alignment for usage inside of CUDA kernels.
-        alignas(16u) TSize m_data[TDim::value];
+        TSize m_data[TDim::value];
```

Comment:
The first one is the right the second waste registers. 

Am 24. November 2015 19:56:41 MEZ, schrieb Benjamin Worpitz <notifications@github.com>:
>> @@ -511,8 +511,7 @@ namespace alpaka
>>          }
>>  
>>      private:
>> -        // 16 Byte alignment for usage inside of CUDA kernels.
>> -        alignas(16u) TSize m_data[TDim::value];
>> +        TSize m_data[TDim::value];
>
>Is it enough to align at `ALPAKA_OPTIMAL_ALIGNMENT_SIZE(TSize)` or
>would it be better to align at
>`ALPAKA_OPTIMAL_ALIGNMENT_SIZE(TSize[TDim::value])`.
>For a `Vec<DimInt<4>, float>` the first version would align at 8 bytes
>while the second one would align at 16 bytes. I think the second one is
>better because it allows vector instructions to be used (on CPU as well
>as on the
>[GPU](http://devblogs.nvidia.com/parallelforall/cuda-pro-tip-increase-performance-with-vectorized-memory-access/)).
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/ComputationalRadiationPhysics/alpaka/pull/103/files#r45777957

-- 
Diese Nachricht wurde von meinem Android-Mobiltelefon mit K-9 Mail gesendet.